packages/documentation/src/pages/getting-started/common-tasks/new-unit-test.mdx

```diff
@@ -132,47 +153,148 @@ fillDate(
+```js
+// SimpleLoginForm.js
+import React from 'react';
+const SimpleLoginForm = ({ onSubmit }) => {
+  const [error, setError] = React.useState('');
+  function handleSubmit(event) {
+    event.preventDefault();
+    const {
+      usernameInput: { value: username },
+      passwordInput: { value: password },
+    } = event.target.elements;
+    if (!username) {
+      setError('username is required');
+    } else if (!password) {
+      setError('password is required');
+    } else {
+      setError('');
+      onSubmit({ username, password });
+    }
+  }
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit}>
+        <div>
+          <label htmlFor="usernameInput">Username</label>
+          <input id="usernameInput" />
+        </div>
+        <div>
+          <label htmlFor="passwordInput">Password</label>
+          <input id="passwordInput" type="password" />
+        </div>
+        <button type="submit">Submit</button>
+      </form>
+      {error ? <div role="alert">{error}</div> : null}
+    </div>
+  );
+};
+export default SimpleLoginForm;
+```
 
 ```js
+// SimpleLoginForm.unit.spec.jsx
 import React from 'react';
 import { expect } from 'chai';
 import sinon from 'sinon';
-import { mount } from 'enzyme';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import SimpleLoginForm from '../../components/SimpleLoginForm';
+describe('my-application', () => {
+  describe('SimpleLoginForm', () => {
+    it('calls onSubmit with the username and password when submit is clicked', () => {
+        const handleSubmit = sinon.spy();
+        const { getByLabelText, getByText } = render(<SimpleLoginForm onSubmit={handleSubmit} />);
+        const user = { username: 'user123', password: 'password123' };
+
+        userEvent.type(getByLabelText(/username/i), user.username);
+        userEvent.type(getByLabelText(/password/i), user.password);
+        userEvent.click(getByText(/submit/i));
+
+        expect(handleSubmit.callCount).to.equal(1); // alternatively `expect(handleSubmit.calledOnce).to.be.true()` works as well
+        expect(handleSubmit.calledWith(user)).to.be.true(); // for more explicit testing we can use `calledWithExactly` in place of `calledWith`
+    });
+  });
+});
+```
+
+We have successfully performed a "Happy Path" test for this component. Let's break down this test:
 
+```js
+// Setup
+...
+const handleSubmit = sinon.spy();
+const { getByLabelText, getByText } = render(<SimpleLoginForm onSubmit={handleSubmit} />);
+const user = { username: 'user123', password: 'password123' };
+...
+```
+
+In setting up this test, we create a mocked submit function and pass that to our form component, then use the [render](https://testing-library.com/docs/react-testing-library/api/#render) function to produce actual DOM nodes. We then, define user data for reuse later on.
```

Comment:
Consider making bulleted lists for these types of paragraphs.

Reply:
Something like this:

```suggestion
1. We mock the `handleSubmit` function.
2. We instantiate the `SimpleLoginForm` component, passing in the mocked `handleSubmit` as a prop.
3. We use the `render` function from RTL to produce actual DOM nodes.
4. We get the RTL query functions from destructuring the return value of `render`. Alternatively, you can utilize screen to access the query functions directly.
4. We define the user data for reuse later in the test.
```

Reply:
- [x] Formatting and phrasing